cmake/QtFunctions.cmake

```diff
@@ -20,7 +20,9 @@ macro(find_qt_package)
         set(MY_OPTIONS_COMPONENTS "")
     endif()
 
-    find_package(Qt6 COMPONENTS ${MY_OPTIONS_COMPONENTS} QUIET)
+    if(NOT QT_DEFAULT_MAJOR_VERSION OR QT_DEFAULT_MAJOR_VERSION STREQUAL "6")
```

Comment:
We may have a problem here, when the user set `QT_DEFAULT_MAJOR_VERSION` to 6, but find_package does not find it. As we check it as `quiet`,  we will go to line 28 (`find_package(Qt5 5.15 ...)`. If Qt5 is found, we may have a mismatch between the user requirement and the actual configuration, without any error or warning.